File: source/aoBlogs2/Models/BlogCommentModel.vb


Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    Public Class BlogCommentModel
        Inherits DbModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Blog Comments"
        Public Const contentTableName As String = "ccBlogComments"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        'instancePropertiesGoHere
        Public Property AllowComments As Boolean
        Public Property Anonymous As Boolean
        Public Property Approved As Boolean
        Public Property articlePrimaryImagePositionId As Integer
        Public Property AuthorMemberID As Integer
        Public Property blogCategoryID As Integer
        Public Property BlogID As Integer
        ' Public Property Copy As String
        Public Property EntryID As Integer
        Public Property FormKey As String
        Public Property imageDisplayTypeId As Integer
        Public Property primaryImagePositionId As Integer
        Public Property Viewings As Integer
        Public Property CopyText As String

    End Class
End Namespace

File: source/aoBlogs2/Models/Db/BlogCommentModel.cs

using Contensive.Models.Db;

namespace Contensive.Blog.Models {
    public class BlogCommentModel : Contensive.Models.Db.DbBaseModel { 
        // 
        // ====================================================================================================
        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Blog Comments", "ccBlogComments", "default", false);
        // 
        // ====================================================================================================
        // -- instance properties
        // instancePropertiesGoHere
        //
        /// <summary>
        /// displayed only if approved. if blog.autoApproveComments then it is automatic, else must be manually approved
        /// </summary>
        public bool approved { get; set; }
        /// <summary>
        /// The blog for this comment
        /// </summary>
        public int blogId { get; set; }
        /// <summary>
        /// the blog entry for this comment
        /// </summary>
        public int entryId { get; set; }
        /// <summary>
        /// no idea what this does. Saved from comment proces
        /// </summary>
        public string formKey { get; set; }
        /// <summary>
        /// The text comment 
        /// </summary>
        public string copyText { get; set; }

    }
}